R/class-color.R

```diff
@@ -0,0 +1,84 @@
+#' R6 class for a Workbook Color
+#'
+#' A color
+#'
+#' @export
+wbColor <- R6::R6Class(
+  "wbColor",
+
+  public = list(
+
+    #' @field auto auto
+    auto = NULL,
+
+    #' @field indexed indexed
+    indexed = NULL,
+
+    #' @field rgb rgb
+    rgb = NULL,
+
+    #' @field theme theme
+    theme = NULL,
+
+    #' @field tint tint
+    tint = NULL,
+
+    #' @description
+    #' Creates a new `wbColor` object
+    #' @param name A name of a color known to R
+    #' @param auto A boolean.
+    #' @param indexed An indexed color values.
+    #' @param rgb A rgb color as ARGB hex value "FF000000".
```

Comment:
Maybe `hex` as the param name instead of `rgb`? I'd imagine we could have `rgb` to be a 3-length vector of integers and check for a `name`, `hex`, and `rgb` value as possible entries.

Reply:
If all are `NULL` can we default to `name = "black"`?  It might be more simple to call `wb_color()` as the defaults for other arguments in functions.

Reply:
Regarding the first, in open xml hex values are `<color rgb="..."/>`, therefore it seems natural, but I agree, calling it hex on our end might be a better idea. Allowing rgb values as numeric vector of three it not something I'm really interested in.

default black might be a good idea, might as well provide a default color option.